### ISSUE TITLE
Tempo-Sync lock

### ISSUE THREAD

**Issue by Andreya-Autumn** [surge-synthesizer/surge#7380]
You know how many reverbs have a lock switch on the wet/dry param which prevents it from changing when you page through presets? User nicokaniac on Discord had the idea that it could be fun to have such a switch but that locks tempo-sync engaged on all relevant params, effectively rounding the saved value to a nearby tempo-synced one. 

It's an interesting idea. My thinking is for some parameters (EG's, Formula and MSEG Rate, maybe others too) it would be such a toss-up whether the tempo-sync rounding would sound cool or just break the patch completely that those should probably be omitted. But for delay times, sequencer rates, and most LFO rates, it might actually work great.

**Comment by RustoMCSpit** [comment]
this would be very useful for locking surge to audio input mode
so latch and audio in

**Comment by RustoMCSpit** [comment]
Discord discussion

for an audio only use case, it's about experimenting and going through them fast
seeing which ones work
which dont
it's fine if many are broken
it doesnt effect anything
the only technical limitation is that youll have to tell all the skin designers they need to incorporate a "locked" color
does that make sense

it makes sense to categorise what can and cant be used with audio in
even if audio in isnt the default oscillator
so an audio compatible
tag
and then people can param lock to audio and latch
but obviously param locking can be done with anything
and this should be the only "compatible" tag
i cant imagine a use case needing another
and they should be able to take multiple tags
so Bass; Audio Compatible; Grungy
but again, audio compatible doesnt mean it's by default having an audio oscillator
that would just be the Audio tag
so Audio isnt Audio Compatible
and vice versa

**Comment by RustoMCSpit** [comment]
youll have to tell all the skin designers they need to incorporate a "locked" color

**Comment by mkruselj** [comment]
> it makes sense to categorise what can and cant be used with audio in

I am not sure this is true. Who's to say what is supposed to work with what, it is up to the user to explore and find out. Experimentation is key to sounding different, rather than being a copycat. Also the suggested wording of such a tag is confusing as well. So, overall I don't think this is a particularly useful idea regarding tagging.

**Comment by RustoMCSpit** [comment]
> > it makes sense to categorise what can and cant be used with audio in
> 
> I am not sure this is true. Who's to say what is supposed to work with what, it is up to the user to explore and find out. Experimentation is key to sounding different, rather than being a copycat. Also the suggested wording of such a tag is confusing as well. So, overall I don't think this is a particularly useful idea regarding tagging.

some patches literallt do not produxe sound in audio mode

**Comment by RustoMCSpit** [comment]
> > it makes sense to categorise what can and cant be used with audio in
> 
> I am not sure this is true. Who's to say what is supposed to work with what, it is up to the user to explore and find out. Experimentation is key to sounding different, rather than being a copycat. Also the suggested wording of such a tag is confusing as well. So, overall I don't think this is a particularly useful idea regarding tagging.

what i meant is that some patches literally do not produce sound at all in audio latch mode. as long as it is capable of producing sound, it is "audio compatible". any patch that lets surge effectively work like an effect over audio rather than an instrument (not literally but you know what i mean) would be tagged as such, regardless of how "nice" it sounds

**Comment by RustoMCSpit** [comment]
>  Also the suggested wording of such a tag is confusing as well. So, overall I don't think this is a particularly useful idea regarding tagging.

the wording can change, it was just an example. "audio conducive"? "works with audio"? "latch friendly"? whatever you think is easy for people to understand

**Comment by mkruselj** [comment]
The more I think about this issue the more it crystallizes to me that this is such an utter can of worms that we're better off not doing it at all. About the only thing that makes sense to lock across patches seems to be pitch bend range. Nothing else makes sense, not even whole effects - because then what do you do about modulated FX parameters? If a patch you're switching to modulates a parameter that in the prior patch (on which you locked the whole FX grid) was discrete and not continuous, you get a gnarly situation...

**Comment by Andreya-Autumn** [comment]
Agreed. Let's just close this for now. Can always reopen if we come up with some genius idea in the future.
